File: src/Samvaad/services/polyglot/Hello.java
public class Hello {
    public static void main(String[] args) {
        if (args.length < 1) {
            System.out.println("{\"error\": \"No arguments provided\"}");
            return;
        }

        String input = args[0];
        String name = "Unknown";
        
        // Simple extraction for showcase (assumes {"name":"..."})
        if (input.contains("\"name\"")) {
            int start = input.indexOf("\"name\"");
            int colon = input.indexOf(":", start);
            int quote1 = input.indexOf("\"", colon);
            int quote2 = input.indexOf("\"", quote1 + 1);
            if (quote1 != -1 && quote2 != -1) {
                name = input.substring(quote1 + 1, quote2);
            }
        }
        
        // Output raw JSON
        System.out.println("{");
        System.out.println("  \"status\": \"success\",");
        System.out.println("  \"lang\": \"Java\",");
        System.out.println("  \"greeting\": \"Hello " + name + " from Java!\",");
        System.out.println("  \"received_data\": " + input);
        System.out.println("}");
    }
}

File: src/Samvaad/services/polyglot/hello.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

func main() {
	if len(os.Args) < 2 {
		output, _ := json.Marshal(map[string]string{"error": "No arguments provided"})
		fmt.Println(string(output))
		return
	}

	var data map[string]interface{}
	err := json.Unmarshal([]byte(os.Args[1]), &data)
	
	if err != nil {
		output, _ := json.Marshal(map[string]string{"error": err.Error()})
		fmt.Println(string(output))
		return
	}

	name := "Unknown"
	if n, ok := data["name"].(string); ok {
		name = n
	}

	response := map[string]interface{}{
		"status":        "success",
		"lang":          "Go",
		"greeting":      fmt.Sprintf("Hello %s from Go!", name),
		"received_data": data,
	}

	output, _ := json.Marshal(response)
	fmt.Println(string(output))
}
